HeatStroke/GOComponents/GameObject.cpp

```diff
@@ -113,6 +120,66 @@ void GameObject::DeleteAllComponents()
 	m_mComponentMap.clear();
 }
 
+
+void GameObject::AddChild(GameObject* p_pChild)
+{
+	if (p_pChild != nullptr)
+	{
+		m_mChildMap.insert(std::pair<std::string, GameObject*>(p_pChild->GetGUID(), p_pChild));
+		p_pChild->SetParent(this);
+		m_Transform.AddChild(&(p_pChild->GetTransform()));
```

Comment:
It looks like your SetParent() method is the problem. SetParent() calls RemoveChild() on the previous parent, which then calls SetParent() again, which calls RemoveChild() again, etc., until parent is definitely a nullptr. I had mentioned you watch out for this. You will have to resolve it before it can be accepted.